coalib/coala.py

```diff
@@ -59,26 +59,56 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
-            show_bears(local_bears,
-                       global_bears,
-                       args.show_description or args.show_details,
-                       args.show_details,
-                       console_printer)
-
+                get_all_bears)
+            all_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                if args.filter_by is None:
+                    args.filter_by = []
+                filter = ['language']
+                filter.extend(args.filter_by_language)
+                args.filter_by.append(filter)
+            if args.filter_by:
+                # Each iteration of the following loop applies
+                # filters one by one provided as arguments
+                for filter_by in args.filter_by:
+                    from coalib.parsing.Filter import Filter
+                    filter = filter_by[0]
+                    filter_args = filter_by[1:]
+                    try:
+                        all_bears = Filter.get_filtered_bears(
+                            filter, filter_args, all_bears)
+                    except Exception as ex:
+                        console_printer.print(ex)
+                        return 255
```

Comment:
@Makman2 I'm returning 255 as error code in case the first argument of filter-by (type of filter) is not in the list. I'm not sure if there is a meaning of each return code in coala conventions. I tried to find if there's a list which describe the meaning of return codes, or list of constants/enums assigning a constant variable with return code, but didn't find it.
Can you please confirm if this return code is valid or any list of constant return code variable is available ?
Or if any other improvements are required, or is good to merge ?